### DISCUSSION TITLE
Legacy - editing migrated games

### DISCUSSION THREAD

**Discussion by dakom** [ji-devs/ji-cloud#2305] [General]
Here is a list of all the games that were added to release: 

https://storage.googleapis.com/ji-cloud-legacy-eu-001/17000-report/create.txt

and here is the same thing for sandbox:

https://storage.googleapis.com/ji-cloud-legacy-eu-001/17000-report/create-sandbox.txt

the structure is:

[jig id] [tt game id] [tt hash id]

Once the game id is known, the original game JSON is at `https://legacy.jicloud.org/games/{game_id}/json/game.json`
## For developers

1. `cd frontend/apps`
2. `cargo make local-main module/legacy/play`
3. visit `http://localhost:4104/module/legacy/play/debug/debug?game_id={GAME_ID}&slide_index={SLIDE_INDEX}` where `GAME_ID` is the `tt game id` in the file above

## For content creators

The storage is on Google Cloud Storage and is proxied by our CDN via Fastly as legacy.jicloud.org but it is not automatically purged by cloud functions (if games are edited, they may require manually purging via the Fastly dashboard)

The bucket name is `ji-cloud-legacy-eu-001` and it’s owned by the `Ji Cloud` project (not sandbox). Inside of there is a `games` folder. You have to find the game according to either of the above docs, and then go into the corresponding `tt game id` folder. Inside of this folder will be a `json` and `media` folder. Inside of the `json` folder is the original `game.json` downloaded from Tiny Tap, and then a `slides` folder **with the converted Jigzi-format JSON for each slide**. Although the order is technically determined by the Jigzi database (they are modules in a jig), unless they were manually changed, they will also match the order in the original `game.json`.

**These slides are the only thing to edit (they correspond to the modules)**. Jig editing is handled like any other jig, via the database. There is never a reason to edit the `game.json`, it's merely there as a reference and to make updating the modules easier (i.e. if we want to re-map things or whatever, without hitting TT's server)

Note that editing and replacing files is though Cloud Storage (either through the browser at google cloud console, cli tools like gsutil, or gui tools like Cyberduck).

The Legacy modules in the jigs do nothing more than load these JSON and media, there is nothing to change in the Jig or module itself.

## be very careful - there is no history for deleted or changed files!

As an example, if I wanted to edit jig id 8fc2c5ca-69bb-11ec-bb37-ab5b27d6a825 on release or d11459b4-6f13-11ec-8e86-d30d69e7694c on sandbox:

1. See they both correspond to the game id `1690`. 
2. Navigate to the cloud storage, the original tiny tap JSON at `ji-cloud-legacy-eu-001/games/1690/json/game.json` and the various Jigzi slides in `ji-cloud-legacy-eu-001/games/1690/json/slides/`.
3. Edit these slide files (through the browser directly or downloading/replacing/uploading), and then see them changed immediately via the web links at, say https://storage.googleapis.com/ji-cloud-legacy-eu-001/games/1690/json/slides/photo1.json - and I would also see them at the CDN-powered links at https://legacy.jicloud.org/games/1690/json/slides/photo1.json once the Fastly cache is purged.

local dev url after `cargo make local-main module/legacy/play` is `http://localhost:4104/module/legacy/play/debug/debug?game_id=GAME_ID&slide_index=SLIDE_INDEX`

comparative Ji Tap player is `https://jitap.net/media/webplayer/webplayer.html?structureJson=https://jitap.net/store/api/album/GAME_ID/structure`

**Comment by corinnewo** [comment]
My notes on how to do this: https://www.dropbox.com/s/zyqo3ayumwyg0ew/How%20to%20edit%20Ji%20Tap%20games%20in%20Jigzi.docx?dl=0

**Comment by corinnewo** [comment]
Corinne's updates to the instructions:
1. IML file for codes: https://docs.google.com/spreadsheets/d/1Oas6xs4dAUMaFeL-rGfuuzHx0L9UNCr7HT20ocn_4Rs/edit?usp=sharing, use the code in the right-most column to look up in GCP.
2. In Fastly, the URL to purge is for example, https://legacy.jigzi.org/transcode/games/18980/json/slides/photo44.json and not https://storage.googleapis.com/ji-cloud-legacy-eu-001/transcode/games/18980/json/slides/photo44.json
3. Finding the slide is not straightforward - eg slide 37 in the JIG corresponded with photo44! Need to try and identify the slide by looking at the media folder and trying to identify the slide. Then can see which number JSON it is.
